src/coreclr/src/jit/treelifeupdater.cpp

```diff
@@ -139,31 +143,31 @@ void TreeLifeUpdater<ForCodeGen>::UpdateLifeVar(GenTree* tree)
                         {
                             // We repeat this call here and below to avoid the VarSetOps::IsMember
                             // test in this, the common case, where we have no deadTrackedFieldVars.
-                            if (fldVarDsc->lvIsInReg())
+                            if (isInReg)
                             {
                                 if (isBorn)
                                 {
                                     compiler->codeGen->genUpdateVarReg(fldVarDsc, tree);
                                 }
                                 compiler->codeGen->genUpdateRegLife(fldVarDsc, isBorn, isDying DEBUGARG(tree));
                             }
-                            else
+                            if (isInMemory)
```

Comment:
Would these changes from `else` to `if (isInMemory)` be better as `else if (isInMemory)` with another `else` with a `noway_assert()`?

Reply:
No; the whole point of this change is that a variable can now be valid both in register and in memory, because when we "write-thru" a definition of an EH exposed variable it can remain live in the register as well.